### PR TITLE
Check if input segmentation is binary for QC report

### LINKED ISSUE #3434: Binary segmentation is displayed as soft in QC report
### Description

With the new feature in QC report to display soft segmentations, currently, the same colormap is used for binary segmentations and soft segmentations. Binarization was removed (see [here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3418#issuecomment-860766982))
It wasn't a problem, for the majority of the images, the binary segmentation is displayed as binary, but I found a few exceptions: 

image

### Steps to Reproduce
0. Install SCT  `git-master-a1c5acbbb1f76d80cbed1715df58f98e7f08ee51`
1. Run `sct_qc `for subject  sub-mpicbs02_T1w.nii.gz in `data.neuro.polymtl.ca:datasets/spine-generic-processed`
```
sct_qc -i /spine-generic-processed/sub-mpicbs02/anat/sub-mpicbs02_T1w.nii.gz -s /spine-generic-processed/derivatives/labels/sub-mpicbs02/anat/sub-mpicbs02_T1w_seg-manual.nii.gz -p sct_deepseg_sc
```
This is not the case when calling `sct_deepseg_sc`on the same image. 
Here are the headers of both images.

<details>
  <summary>sub-mpicbs02_T1w_seg-manual.nii.gz </summary>

  ``` python
sizeof_hdr     348
data_type      FLOAT64
dim0           3
dim1           192
dim2           260
dim3           320
dim4           1
dim5           1
dim6           1
dim7           1
vox_units      mm
time_units     Unknown
datatype       64
nbyper         8
bitpix         64
pixdim0        0.000000
pixdim1        0.999999
pixdim2        1.000000
pixdim3        1.000000
pixdim4        1.000000
pixdim5        1.000000
pixdim6        1.000000
pixdim7        1.000000
vox_offset     352
cal_max        0.0000
cal_min        0.0000
scl_slope      1.000000
scl_inter      0.000000
phase_dim      0
freq_dim       0
slice_dim      0
slice_name     Unknown
slice_code     0
slice_start    0
slice_end      0
slice_duration 0.000000
time_offset    0.000000
intent         Unknown
intent_code    0
intent_name
intent_p1      0.000000
intent_p2      0.000000
intent_p3      0.000000
qform_name     Scanner Anat
qform_code     1
qto_xyz:1      -0.999657  0.000000  -0.026174  99.629417
qto_xyz:2      -0.000000  1.000000  0.000000  -106.576935
qto_xyz:3      -0.026174  -0.000000  0.999657  -207.853012
qto_xyz:4      0.000000  0.000000  0.000000  1.000000
qform_xorient  Right-to-Left
qform_yorient  Posterior-to-Anterior
qform_zorient  Inferior-to-Superior
sform_name     Scanner Anat
sform_code     1
sto_xyz:1      -0.999657  0.000000  -0.026177  99.629417
sto_xyz:2      0.000000  1.000000  0.000000  -106.576935
sto_xyz:3      -0.026177  0.000000  0.999657  -207.853012
sto_xyz:4      0.000000  0.000000  0.000000  1.000000
sform_xorient  Right-to-Left
sform_yorient  Posterior-to-Anterior
sform_zorient  Inferior-to-Superior
file_type      NIFTI-1+
file_code      1
descrip
aux_file
```

</details>

<details>
  <summary>sub-mpicbs02_T1w_seg.nii.gz</summary>

``` python
filename       sub-mpicbs02_T1w_seg.nii.gz

sizeof_hdr     348
data_type      UINT8
dim0           3
dim1           192
dim2           260
dim3           320
dim4           1
dim5           1
dim6           1
dim7           1
vox_units      Unknown
time_units     Unknown
datatype       2
nbyper         1
bitpix         8
pixdim0        0.000000
pixdim1        0.999999
pixdim2        1.000000
pixdim3        1.000000
pixdim4        1.000000
pixdim5        1.000000
pixdim6        1.000000
pixdim7        1.000000
vox_offset     352
cal_max        0.0000
cal_min        0.0000
scl_slope      1.000000
scl_inter      0.000000
phase_dim      0
freq_dim       0
slice_dim      0
slice_name     Unknown
slice_code     0
slice_start    0
slice_end      0
slice_duration 0.000000
time_offset    0.000000
intent         Unknown
intent_code    0
intent_name
intent_p1      0.000000
intent_p2      0.000000
intent_p3      0.000000
qform_name     Aligned Anat
qform_code     2
qto_xyz:1      -0.999657  0.000000  -0.026174  99.629417
qto_xyz:2      -0.000000  1.000000  0.000000  -106.576935
qto_xyz:3      -0.026174  -0.000000  0.999657  -207.853012
qto_xyz:4      0.000000  0.000000  0.000000  1.000000
qform_xorient  Right-to-Left
qform_yorient  Posterior-to-Anterior
qform_zorient  Inferior-to-Superior
sform_name     Aligned Anat
sform_code     2
sto_xyz:1      -0.999657  0.000000  -0.026177  99.629417
sto_xyz:2      0.000000  1.000000  0.000000  -106.576935
sto_xyz:3      -0.026177  0.000000  0.999657  -207.853012
sto_xyz:4      0.000000  0.000000  0.000000  1.000000
sform_xorient  Right-to-Left
sform_yorient  Posterior-to-Anterior
sform_zorient  Inferior-to-Superior
file_type      NIFTI-1+
file_code      1
descrip
aux_file
```

</details>

Maybe the problem is with the manual segmentation, I will investigate if the other cases are also manual segmentation!

### Suggestion
* We can add a flag to display the soft segmentation on demand.
OR
* Detect if the input segmentation is soft or binary and binarize in the second case.

### PR DESCRIPTION
**Pull request by sandrinebedard**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

This PR addresses #3434. For generating QC report of spinal cord segmentation, when resampling a binary segmentation (with linear interpolation), it introduces non-binary values. Binarization was removed in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3418 to be able to display segmentations with a colormap. A threshold of 0.5 was applied. Most binary segmentations were displayed correctly in the QC report, but there were a few exceptions.
 
This PR includes :
* Check if input image is binary
* If the image is binary, binarize at 0.5
* Else, apply a threshold at 0.5

To test the fix, I suggest to run `sct_qc` on `/spine-generic-processed/derivatives/labels/sub-mpicbs02/anat/sub-mpicbs02_T1w_seg-manual.nii.gz` from `data.neuro.polymtl.ca:datasets/spine-generic-processed` which was problematic.

## Linked issues

Resolves #3434

### CODE REVIEW

**Review by jcohenadad — APPROVED** [review]
LGTM! thank you for catching the issue and fixing it

File: spinalcordtoolbox/reports/slice.py [hunk 1]
```diff
@@ -340,6 +340,8 @@ def _resample_slicewise(self, image, p_resample, type_img, image_ref=None):
         dict_interp = {'im': 'spline', 'seg': 'linear'}
         # Create nibabel object
         nii = Nifti1Image(image.data, image.hdr.get_best_affine())
+        # Check if input image is binary
+        is_binary = np.isin(nii.get_data(), [0, 1]).all()
```
Comment (joshuacwnewton) [inline spinalcordtoolbox/reports/slice.py:344]: Hmm, will this work if the image is float? I'm not sure off the top of my head how `np.isin` works for float/int comparison.

Comment (joshuacwnewton) [reply spinalcordtoolbox/reports/slice.py:344]: Also, one other small small nitpick -- Do we need to do this check here?

If `is_binary` is only used in a single place, maybe we could just have this directly inside the if/else block.

Comment (joshuacwnewton) [reply spinalcordtoolbox/reports/slice.py:344]: Eh, honestly, this is too much of a nitpick to be worth fixing. Scratch my recommendation. :P

**Review by joshuacwnewton — APPROVED** [review]

**Pull request merged**